src/coordination/coordinator_state_manager.cpp

```diff
@@ -27,44 +27,61 @@ using nuraft::srv_state;
+constexpr std::string_view kClusterConfigKey = "cluster_config";  // Key prefix for cluster_config durability
+
+constexpr std::string_view kServerStateKey = "server_state";  // Key prefix for server state durability
+constexpr std::string_view kVotedFor = "voted_for";
+constexpr std::string_view kTerm = "term";
+constexpr std::string_view kElectionTimer = "election_timer";
+
+constexpr std::string_view kStateManagerDurabilityVersionKey = "state_manager_durability_version";
+constexpr int kActiveStateManagerDurabilityVersion = 1;
+
 }  // namespace
 
-CoordinatorStateManager::CoordinatorStateManager(CoordinatorInstanceInitConfig const &config, LoggerWrapper logger)
-    : my_id_(static_cast<int>(config.coordinator_id)),
-      cur_log_store_(cs_new<CoordinatorLogStore>()),
+CoordinatorStateManager::CoordinatorStateManager(CoordinatorStateManagerConfig const &config, LoggerWrapper logger)
+    : my_id_(static_cast<int>(config.coordinator_id_)),
+      cur_log_store_(cs_new<CoordinatorLogStore>(config.durability_store_, logger)),
       logger_(logger),
-      kv_store_(config.durability_dir) {
+      state_manager_durability_(config.state_manager_durability_dir_) {
   auto const c2c =
-      CoordinatorToCoordinatorConfig{config.coordinator_id, io::network::Endpoint("0.0.0.0", config.bolt_port),
-                                     io::network::Endpoint{"0.0.0.0", static_cast<uint16_t>(config.coordinator_port)}};
-  my_srv_config_ = cs_new<srv_config>(config.coordinator_id, 0, c2c.coordinator_server.SocketAddress(),
-                                      nlohmann::json(c2c).dump(), false);  // non-resolved IP in SocketAddress
+      CoordinatorToCoordinatorConfig{config.coordinator_id_, io::network::Endpoint("0.0.0.0", config.bolt_port_),
+                                     io::network::Endpoint{"0.0.0.0", static_cast<uint16_t>(config.coordinator_port_)}};
+  my_srv_config_ = cs_new<srv_config>(config.coordinator_id_, 0, c2c.coordinator_server.SocketAddress(),
+                                      nlohmann::json(c2c).dump(), false);
 
   cluster_config_ = cs_new<cluster_config>();
   cluster_config_->get_servers().push_back(my_srv_config_);
+
+  int version{0};
+  auto maybe_version = state_manager_durability_.Get(kStateManagerDurabilityVersionKey);
+  if (maybe_version.has_value()) {
+    version = std::stoi(maybe_version.value());
+  } else {
+    logger_.Log(
+        nuraft_log_level::INFO,
+        fmt::format("Assuming first start of state manager with durability as version is missing, storing version {}.",
+                    kActiveStateManagerDurabilityVersion));
+    MG_ASSERT(state_manager_durability_.Put(kStateManagerDurabilityVersionKey,
+                                            std::to_string(kActiveStateManagerDurabilityVersion)),
+              "Failed to store version to disk");
+    version = kActiveStateManagerDurabilityVersion;
+  }
+
```

Comment:
Where do we have code for bumping versioning? Let's say you read version v1 but are currently at v2, where is this handling?

Reply:
We can add it once we need it, I wouldn't add that code now. It would be a new function and then we define what we need. If I add it now, it sounds like premature optimization (because we don't know what we will need)

Reply:
change version to be enum